public/pages/workflow_detail/workspace/workspace.tsx

```diff
@@ -44,6 +52,24 @@ const edgeTypes = { customEdge: DeletableEdge };
 export function Workspace(props: WorkspaceProps) {
   const dispatch = useAppDispatch();
 
+  // Visual/JSON toggle states
+  const [visualSelected, setVisualSelected] = useState<boolean>(true);
+  const [jsonSelected, setJsonSelected] = useState<boolean>(false);
+  function toggleSelection(): void {
+    setVisualSelected(!visualSelected);
+    setJsonSelected(!jsonSelected);
+  }
```

Comment:
nit: I might don't have enough context on what the visualSelected and jsonSelected variable mean here, but it seems like they are set to complementary values. I'm ok with setting it this way but just a nit comment if they are mutually exclusive we can derive one from the other

Reply:
Good point! Removed to just be one var.